src/cpp/monitor/events/Event.cpp

```diff
@@ -24,6 +24,24 @@
 namespace rstudio {
 namespace monitor {
 
+Event::Event(int scope,
+             int id,
+             const std::string& data,
+             const std::string& username,
+             PidType pid,
+             boost::posix_time::ptime timestamp)
+   : empty_(false),
+     scope_(scope),
+     id_(id),
+     pid_(pid),
+     timestamp_(core::date_time::millisecondsSinceEpoch(timestamp))
+{
+   ::memset(&username_, 0, kMaxEventDataSize+1);
```

Comment:
Should we check and warn if we're truncating the username / data here? I also worry that we would fail to write the null terminator in these cases.

Reply:
Events carry very little data. Digits or username only.

Reply:
I still think it's worth validating this is true, since IIUC we do construct these events from somewhat arbitrary inputs; e.g. in `eventFromJson()`:

https://github.com/rstudio/rstudio-pro/blob/4caacd464bb703d5741863fea7b6a530ca7ad045/src/cpp/monitor/http/Server.cpp#L118-L134

https://github.com/rstudio/rstudio-pro/blob/4caacd464bb703d5741863fea7b6a530ca7ad045/src/cpp/monitor/events/EventUtils.cpp#L33-L58

Reply:
Warn log added